shell.nix

```diff
@@ -7,21 +7,37 @@ pkgs.mkShell {
   name = "shell";
   buildInputs =
     [
-      pkgs.buildPackages.cabal-install
-      pkgs.buildPackages.gcc
-      pkgs.buildPackages.ghc
-      pkgs.buildPackages.pkg-config
-      pkgs.buildPackages.sbt
-      pkgs.buildPackages.scala
-      pkgs.buildPackages.verilator
+      pkgs.gcc
+      pkgs.pkg-config
+      pkgs.sbt
+      pkgs.scala
+      pkgs.verilator
+
+      # Haskell toolchain
+      pkgs.cabal-install
+      # pkgs.haskell.compiler.ghc90
+      # pkgs.haskell.compiler.ghc92
+      pkgs.haskell.compiler.ghc94
+      # pkgs.haskell.compiler.ghc96
```

Comment:
Since Nix isn't used on CI it doesn't matter too much, is 9.4 the one we should be looking at "mainly" then?